rust/kcl-lib/src/std/constraints.rs

```diff
@@ -476,17 +480,152 @@ pub async fn coincident(exec_state: &mut ExecState, args: Args) -> Result<KclVal
+                                ) => {
+                                    let point = DatumPoint::new_xy(
+                                        point_x.to_constraint_id(range)?,
+                                        point_y.to_constraint_id(range)?,
+                                    );
+                                    let line_segment = LineSegment::new(
+                                        DatumPoint::new_xy(sx.to_constraint_id(range)?, sy.to_constraint_id(range)?),
+                                        DatumPoint::new_xy(ex.to_constraint_id(range)?, ey.to_constraint_id(range)?),
+                                    );
+                                    let constraint = SolverConstraint::PointLineDistance(point, line_segment, 0.0);
+
+                                    #[cfg(feature = "artifact-graph")]
+                                    let constraint_id = exec_state.next_object_id();
+
+                                    let Some(sketch_state) = exec_state.sketch_block_mut() else {
+                                        return Err(KclError::new_semantic(KclErrorDetails::new(
+                                            "coincident() can only be used inside a sketch block".to_owned(),
+                                            vec![args.source_range],
+                                        )));
+                                    };
+                                    sketch_state.solver_constraints.push(constraint);
+                                    #[cfg(feature = "artifact-graph")]
+                                    {
+                                        let constraint = crate::front::Constraint::Coincident(Coincident {
+                                            points: vec![unsolved0.object_id, unsolved1.object_id],
+                                        });
+                                        sketch_state.sketch_constraints.push(constraint_id);
+                                        track_constraint(constraint_id, constraint, exec_state, &args);
+                                    }
+                                    Ok(KclValue::none())
+                                }
+                                _ => Err(KclError::new_semantic(KclErrorDetails::new(
+                                    "Line segment endpoints must be sketch variables for point-segment coincident constraint".to_owned(),
+                                    vec![args.source_range],
+                                ))),
+                            }
+                        }
+                        _ => Err(KclError::new_semantic(KclErrorDetails::new(
+                            "Point coordinates must be sketch variables for point-segment coincident constraint"
+                                .to_owned(),
+                            vec![args.source_range],
+                        ))),
+                    }
+                }
+                // Line-Point case: same as Point-Line, just reversed
```

Comment:
If the two cases are symmetric (order of line-point or point-line), you should be able to delete this entire second branch. Rust lets you do:

```rust
match e {
  pat1 { foo, bar } | pat2 { foo, bar } => { ... }
}
```

...if bindings in the two patterns are the same. I.e. both pat1 and pat2 bind `foo` and `bar` and they're the same type in both patterns.

Reply:
Thank you! f5e7da4d7ae24e2506aac0d0d57977d4f0196a96